mapper/user.go

```diff
@@ -24,3 +24,87 @@ func CreateUser(ctx context.Context, user model.User) error {
 
 	return db.Create(&userTable).Error
 }
+
+func DeleteUser(ctx context.Context, user model.User) error {
+	db := database.GetDefaultDB()
+	return db.Delete(&model.UserTable{Account: user.Account}).Error
+}
+
+func UpdateUser(ctx context.Context, user model.User) error {
+	db := database.GetDefaultDB()
+	var hashedPassword string
+	if user.Password != nil {
+		var err error
+		hashedPassword, err = utils.GetHashedPassword(*user.Password, argon2id.DefaultParams)
+		if err != nil {
+			return err
+		}
+	} else {
+		hashedPassword = ""
+	}
+
+	userRow := model.UserTable{
+		Account:        user.Account,
+		Name:           user.Name,
+		HashedPassword: hashedPassword,
+		Roles:          user.Roles.ToPQArray(),
+		Email:          user.Email,
+		Mobile:         user.Mobile,
+	}
+
+	return db.Model(&model.UserTable{Account: userRow.Account}).Updates(userRow).Error
+}
+
+type UserOption struct {
+	Account *string
+	Email   *string
+	Mobile  *string
+}
+
+func GetUserInfo(ctx context.Context, option UserOption) (*model.UserInfo, error) {
+	db := database.GetDefaultDB()
+	account := ""
+	if option.Account != nil {
+		account = *option.Account
+	}
+	var user model.UserTable
+	err := db.Where(&model.UserTable{Account: account, Email: option.Email, Mobile: option.Mobile}).First(&user).Error
+	if err != nil {
+		return nil, err
+	}
+	return &model.UserInfo{
+		Account:  user.Account,
+		Name:     user.Name,
+		Roles:    model.PQArray2Roles(&user.Roles),
+		Email:    user.Email,
+		CreateAt: user.CreateAt,
+		UpdateAt: user.UpdateAt,
+	}, nil
+}
+
+type FuzzyQuery struct {
+	query  string
+	offset int
+	limit  int
+}
+
+func FindUserInfo(ctx context.Context, fq FuzzyQuery) ([]model.UserInfo, error) {
+	db := database.GetDefaultDB()
+	var users []model.UserTable
+	err := db.Where("account LIKE ?", fq.query).Or("name LIKE ?", fq.query).Offset(fq.offset).Limit(fq.limit).Find(&users).Error
+	if err != nil {
+		return nil, err
+	}
+	var userInfos []model.UserInfo
+	for _, user := range users {
+		userInfos = append(userInfos, model.UserInfo{
+			Account:  user.Account,
+			Name:     user.Name,
+			Roles:    model.PQArray2Roles(&user.Roles),
+			Email:    user.Email,
+			CreateAt: user.CreateAt,
+			UpdateAt: user.UpdateAt,
+		})
+	}
+	return userInfos, err
+}
```

Comment:
Should only return `model.User` here, convert of `model.UserInfo` can be done from higher layer